src/mono/mono/metadata/marshal-lightweight.c

```diff
@@ -2446,7 +2446,11 @@ emit_unsafe_accessor_ctor_wrapper (MonoMethodBuilder *mb, MonoMethod *accessor_m
 	MonoClass *in_class = mono_class_is_ginst (target_class) ? mono_class_get_generic_class (target_class)->container_class : target_class;
 	MonoMethod *target_method = mono_unsafe_accessor_find_ctor (in_class, member_sig, target_class, find_method_error);
 	if (!is_ok (find_method_error) || target_method == NULL) {
-		emit_missing_method_error (mb, find_method_error, "constructor");
+		if (mono_error_get_error_code (find_method_error) == MONO_ERROR_GENERIC)
+			mono_mb_emit_exception_for_error (mb, find_method_error);
+		else {
```

Comment:
Only having braces for one of the branches seems odd. They are both a single statement. I would expect them to each have braces or neither.

Reply:
Same below.

Reply:
Removed the redundant curly braces.